opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -216,136 +257,143 @@ impl<R: RuntimeChannel> LogProcessor for BatchLogProcessor<R> {
-                            if let Err(send_error) = channel.send(result) {
-                                otel_debug!(
-                                    name: "BatchLogProcessor.Flush.SendResultError",
-                                    error = format!("{:?}", send_error),
-                                );
-                            }
+                    Ok(BatchMessage::ForceFlush(_sender)) => {
+                        export_with_timeout_sync(remaining_time, exporter.as_mut(), logs.split_off(0), &mut last_export_time);
+
+                        match _sender.send(Ok(())) {
+                            Ok(_) => {}
+                            Err(err) => global::handle_error(LogError::Other(err.into())),
                         }
                     }
-                    // Stream has terminated or processor is shutdown, return to finish execution.
-                    BatchMessage::Shutdown(ch) => {
-                        let result = export_with_timeout(
-                            config.max_export_timeout,
-                            exporter.as_mut(),
-                            &timeout_runtime,
-                            logs.split_off(0),
-                        )
-                        .await;
-
+                    Ok(BatchMessage::Shutdown(_sender)) => {
+                        export_with_timeout_sync(remaining_time, exporter.as_mut(), logs.split_off(0), &mut last_export_time);
                         exporter.shutdown();
-
-                        if let Err(send_error) = ch.send(result) {
-                            otel_debug!(
-                                name: "BatchLogProcessor.Shutdown.SendResultError",
-                                error = format!("{:?}", send_error),
-                            );
+                        match _sender.send(Ok(())) {
+                            Ok(_) => {}
+                            Err(err) => global::handle_error(LogError::Other(err.into())),
                         }
                         break;
                     }
-                    // propagate the resource
-                    BatchMessage::SetResource(resource) => {
+                    Ok(BatchMessage::SetResource(resource)) => {
                         exporter.set_resource(&resource);
                     }
+                    Err(RecvTimeoutError::Timeout) => {
+                        export_with_timeout_sync(config.max_export_timeout, exporter.as_mut(), logs.split_off(0), &mut last_export_time);
+                    }
+                    Err(err) => global::handle_error(LogError::Other(err.into())),
                 }
             }
-        }));
+        });
+
+        let forceflush_timeout = env::var(OTEL_LOGS_FORCEFLUSH_TIMEOUT_NAME)
+            .ok()
+            .and_then(|v| v.parse().map(Duration::from_millis).ok())
+            .unwrap_or(OTEL_LOGS_DEFAULT_FORCEFLUSH_TIMEOUT);
+        let shutdown_timeout = env::var(OTEL_LOGS_SHUTDOWN_TIMEOUT_NAME)
```

Comment:
agree with using timeout for shutdown, but this env variable is not required. Lets accept a timeout in the shutdown() method itself, and if none provided, use 5 sec as default. (5 sec is not from any spec, just my guess of a good default)